### DISCUSSION TITLE
Transition control (slide or fade) for nested carousel

### DISCUSSION THREAD

**Discussion by Michaelgimii** [elementor/elementor#22821] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Please add Transition feature to the new carousel widget. Currently, slide is the default and it is not possible to change it to fade.

### Describe the solution you'd like

Transition control (slide or fade) for nested carousel widget.

**Comment by arekuki** [comment]
+1

**Comment by mikekellneringenious** [comment]
Hi @nicholaszein, what is the best way to get this suggested for a future release?

**Reply by m1ga** [reply]
https://developers.elementor.com/expose-swiper-in-elementor/
you can do it with this simple JS yourself:

```js
swiper = jQuery ( ".swiper" ).data("swiper");
swiper.params.effect = "fade"; 
swiper.params.fadeEffect.crossFade = true;
swiper.destroy(false)
new Swiper(".swiper", swiper.params)
```

**Reply by mikekellneringenious** [reply]
Thank you so much for this @m1ga I am struggling to get this to work, so I will keep reading up on this.

**Reply by m1ga** [reply]
The quickest way to test it is just to paste the code in the devtools:

https://github.com/elementor/elementor/assets/4334997/4805c68a-ff81-465f-9cf8-b108e0653f0d

Then you know the code is working and the rest is up to you wehre you want to implement it

**Reply by jevmedia** [reply]
Hi how do i implement this on a Elementor website?

**Reply by m1ga** [reply]
you add the JS code into your page :shrug: Google: "add custom javascript to elementor"

**Reply by jevmedia** [reply]
Yea I understand, but when I try adding it to the head, the whole code appears as text. Same when I use an html widget.

**Reply by m1ga** [reply]
please read the tutorials that appear when you search for the line above. You are missing the `<script>` block or you have to use WPCode or other plugins that add code to your head. You don't just put the code in there because it is Javascript.

Or add a custom JS file using PHP,... many options, all are described in posts like https://www.wpcrafter.com/add-javascript-to-elementor-websites/ 

and to answer your future question: you'll need to wait for DOMReady and for the slideshow to be initalized, so you'll might need a timeout

**Reply by warrenabayon** [reply]
it goes all white or blank after all the slides play,, can you please fix this

**Reply by mstudioIL** [reply]
@m1ga it is not working for me
I have error "Uncaught TypeError: Cannot read properties of undefined (reading 'params')"
I want the carousel with the containers do have fade effect and not slide

**Reply by m1ga** [reply]
see the comment above `and to answer your future question:`. It's undefined for you, so you run the script too early.

**Reply by mstudioIL** [reply]
I added it on the page with HTML widget using 
```
<script>
swiper = jQuery ( ".swiper" ).data("swiper");
swiper.params.effect = "fade"; 
swiper.params.fadeEffect.crossFade = true;
swiper.destroy(false)
new Swiper(".swiper", swiper.params)
</script>
```
Should I add custom ID so it won't effect other slide on the page?

**Reply by m1ga** [reply]
please read my comment, I'm not doing JS support here. check some basic tutorials or use the comments of the official Elementor post in my first comment

**Reply by mstudioIL** [reply]
OK, thanks

**Reply by DownHouse** [reply]
> https://developers.elementor.com/expose-swiper-in-elementor/ you can do it with this simple JS yourself:
> 
> ```js
> swiper = jQuery ( ".swiper" ).data("swiper");
> swiper.params.effect = "fade"; 
> swiper.params.fadeEffect.crossFade = true;
> swiper.destroy(false)
> new Swiper(".swiper", swiper.params)
> ```

This works, but the content inside the slides, such as buttons and so on, becomes unavailable for mouse events. 
This may be a swiper bug.
